plugins/tableselection/plugin.js

```diff
@@ -1009,6 +1020,13 @@
 						}
 
 						selection.selectRanges( ranges );
+
+						// We need to lock undoManager to consider clearing table and inserting new paragraph as single operation, and have only one undo step (#1816).
+						if ( key === 13 && editor.plugins.enterkey ) {
```

Comment:
Probably the whole handling of undo should be refactored as now native <kbd>Enter</kbd> also produces two steps (works correctly on `master`).

Reply:
It also means that we need one more unit test for native <kbd>Enter</kbd>, this time connected with undo ;)